apps/pollbook/frontend/src/voter_details_screen.test.tsx

```diff
@@ -0,0 +1,212 @@
+});
+
+// Specifies a DOM element to look for and value to enter
+interface InputElementChangeSpec {
+  domElementText: string;
+  domElementRole?: string;
+  newValue: string;
+}
+
+test('name change', async () => {
+  apiMock.setPrinterStatus(true);
+  apiMock.expectGetDeviceStatuses();
+
+  await renderComponent();
+
+  userEvent.click(screen.getButton('Update Name'));
+
+  const nameParts: InputElementChangeSpec[] = [
+    {
+      domElementText: 'First Name',
+      newValue: 'Nabby',
+    },
+    {
+      domElementText: 'Middle Name',
+      newValue: 'Abigail',
+    },
+    {
+      domElementText: 'Last Name',
+      newValue: 'Addams',
+    },
+    {
+      domElementText: 'Suffix',
+      newValue: 'Jr',
+    },
+  ];
+  for (const part of nameParts) {
+    const partInput = await screen.findByRole('textbox', {
+      name: part.domElementText,
+    });
+    userEvent.type(partInput, part.newValue);
+  }
+
+  const nameChangeData: VoterNameChangeRequest = {
+    lastName: 'ADDAMS',
+    firstName: 'NABBY',
+    middleName: 'ABIGAIL',
+    suffix: 'JR',
+  };
+  const expectation = {
+    voterId: mockVoterId,
+    nameChangeData,
+    voterToUpdate: voter,
+  } as const;
+  apiMock.expectChangeVoterName(expectation);
+  const updatedVoter: Voter = { ...voter, ...nameChangeData };
+  const updateButton = screen.getButton('Confirm Name Update');
+  userEvent.click(updateButton);
+
+  apiMock.expectGetVoter(updatedVoter);
+  userEvent.click(await screen.findButton('Return to Voter Details'));
+
+  await waitFor(() => {
+    expect(screen.queryByText('Update Voter Name')).toBeNull();
+  });
+
+  await screen.findByText('NABBY ABIGAIL ADDAMS JR');
+});
+
+test('address change', async () => {
+  apiMock.setPrinterStatus(true);
+  apiMock.expectGetDeviceStatuses();
+  const validStreetInfo: ValidStreetInfo[] = [
+    {
+      streetName: 'MAIN ST',
+      side: 'all',
+      lowRange: 1,
+      highRange: 100,
+      postalCity: 'CONCORD',
+      zip5: '03301',
+      zip4: '1111',
+      district: '',
+    },
+  ];
+  apiMock.expectGetValidStreetInfo(validStreetInfo);
+
+  await renderComponent();
+
+  userEvent.click(screen.getButton('Update Address'));
+
+  await screen.findByRole('heading', { name: 'Update Voter Address' });
+
+  userEvent.click(screen.getByLabelText('Street Name'));
+  userEvent.keyboard('[Enter]');
+
+  const nameParts: InputElementChangeSpec[] = [
+    {
+      domElementText: 'Street Number',
+      newValue: '99',
+    },
+    {
+      domElementText: 'Apartment or Unit Number',
+      newValue: '#789',
+    },
+    {
+      domElementText: 'Address Line 2',
+      newValue: 'LINE 2',
+    },
+  ];
+  for (const part of nameParts) {
+    const partInput = await screen.findByRole(
+      part.domElementRole ?? 'textbox',
+      {
+        name: part.domElementText,
+      }
+    );
+    userEvent.type(partInput, part.newValue);
+  }
+
+  const addressChangeData: VoterAddressChangeRequest = {
```

Comment:
I know you didn't change this part of the code but while you are here can you test the case that if you have an invalid address according to the street names that the address error shows up?